src/shims/unix/linux/epoll.rs

```diff
@@ -415,43 +416,113 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         events_op: &OpTy<'tcx>,
         maxevents: &OpTy<'tcx>,
         timeout: &OpTy<'tcx>,
-    ) -> InterpResult<'tcx, Scalar> {
+        dest: MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
 
-        let epfd = this.read_scalar(epfd)?.to_i32()?;
+        let epfd_value = this.read_scalar(epfd)?.to_i32()?;
         let events = this.read_immediate(events_op)?;
         let maxevents = this.read_scalar(maxevents)?.to_i32()?;
-        let timeout = this.read_scalar(timeout)?.to_i32()?;
+        let timeout = this.read_scalar(timeout)?.to_u32()?;
```

Comment:
According to the docs the timeout is signed. What happens on a negative timeout?

Reply:
Oh right, I haven't handled this. The documentation said if the timeout is -1, it will block indefinitely until it received a notification. I haven't found documentation on behaviour of other negative numbers, but it still seems to block indefinitely for every negative values on my host system (``x86_64-unknown-linux-gnu``).